core/permcache.ts

```diff
@@ -0,0 +1,229 @@
+// Permanent version of the Cache from cache.ts in the same folder
+import { promises as fsp } from "fs";
+import fs from "fs";
+import path from "path";
+
+export function getVSCodeDenoDir(): string {
```

Comment:
I am a little curious, why not use the system cache folder? Or use the vscode_deno extension folder

Reply:
how to get extension folder?

Reply:
![image](https://user-images.githubusercontent.com/15936231/92329198-7103ef80-f098-11ea-9c87-c9b1f17d116f.png)
But you can change it by setting this environment variable

Reply:
@CGQAQ you can get it via `context.extensionPath` in client-side.

get it via env `VSCODE_DENO_EXTENSION_PATH` in server side

https://github.com/denoland/vscode_deno/blob/65a0928ef9a5db1640ce2bd95f899a8c72ba2ff7/client/src/extension.ts#L201-L206

this directory will change every time when the extension is updated

eg.

denoland.vscode_deno-1.0.0
denoland.vscode_deno-1.1.0

Reply:
![image](https://user-images.githubusercontent.com/15936231/92329274-174ff500-f099-11ea-80e5-a769322d20f4.png)

So, this will be decent?